packages/@sanity/form-builder/src/inputs/SelectInput.tsx

```diff
@@ -26,15 +26,34 @@ const SelectInput = React.forwardRef(function SelectInput(
   const validation = markers.filter((marker) => marker.type === 'validation')
   const errors = validation.filter((marker) => marker.level === 'error')
 
+  const itemFromOptionValue = (optionValue) => {
+    const index = Number(optionValue)
+
+    return items[index]
```

Comment:
I might be wrong, but I believe it should be `items[index - 1]` to find the correct item?

Reply:
(Ah, never mind)